models/DataObject/ContextChain/AbstractNode.php

```diff
@@ -0,0 +1,30 @@
+<?php
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Enterprise License (PEL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ * @category   Pimcore
+ * @package    Object
+ *
+ * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ * @license    http://www.pimcore.org/license     GPLv3 and PEL
+ */
+
+namespace Pimcore\Model\DataObject\ContextChain;
+
+abstract class AbstractNode
```

Comment:
@weisswurstkanone 
Imho a context object should always have a subject, similar to Symfony events. 
The subject is then an instance of a Brick, Localized Fields, ... 
This would make the implementation classes a bit leaner, since we don't need getters/setters for inromation like names, ... since they can be obtained by the subject object. Only contextual information which is not included in the subject object needs to be added to the implementation class, such as index information, or the locale and others. 

What do you think?

Reply:
Maybe also a factory should be responsible in creating these elements